autodiff/variable.py

```diff
@@ -1,24 +1,63 @@
 import numpy as np
 from autodiff.utils import get_right_shape
-#from variable import Variable
+
 
 class Variable:
     """ 
-    #TODO: Write some documentation about that.
-    Variable class carry the information flow within the computational graph. 
-    Data attribute, Gradient attribute.
-    Data represents the evaluation point and gradient is the gradient held 
-    by the variable. By default = 1. 
+    Represents a variable, and carries the information flow
+    within the computational graph. Under the current 
+    implementation, variables are scalar valued.
     """
     def __init__(self, val, grad=1.): 
+        """
+        Variables are initialized with a value and a gradient.
+
+         INPUTS
+        =======
+        val: float or int, required.
+            Is the value of the variable.
+
+        grad: float or int, optional. Default value is 1 (the seed).
+            Is the gradient of the variable.
+
+        EXAMPLE
+        =========
+        >>> x = Variable(2.0)
+        """
+
+        # Assure val and grad are correct shape (in preparation for
+        # multivariate implementation)
         self.val = get_right_shape(val)
         #grad = np.ones((len(self.val), ))
         self.grad = get_right_shape(grad)
     
     def __repr__(self):
+        """ When variables are printed, gives both val and grad"""
         return "Value: {}\nGradient: {}".format(self.val, self.grad)
     
     def __add__(self, other):
+        """Implements addition (self + other) between Variables and
```

Comment:
Can you do the same without writing "self+other" explictly. 
"Implements addition between.." or "overload..." are more than enough.

Reply:
And same thing for __mul_ so on, it's better not to say "self, other"